src/main/java/com/google/netpcapanalysis/dao/FrequencyDaoImpl.java

```diff
@@ -0,0 +1,148 @@
+import io.pkts.packet.TCPPacket;
+import io.pkts.packet.UDPPacket;
+import io.pkts.packet.IPPacket;
+import io.pkts.protocol.Protocol;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import com.google.netpcapanalysis.interfaces.dao.PCAPDao;
+import com.google.netpcapanalysis.interfaces.dao.PCAPParserDao;
+import com.google.netpcapanalysis.interfaces.dao.FrequencyDao;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class FrequencyDaoImpl implements FrequencyDao {
+  private ArrayList<PCAPdata> allPCAP; 
+  private ArrayList<PCAPdata> finalFreq;
+  private HashMap<String, PCAPdata> finalMap;
+
+  private String myip = "";
+  private boolean first = true;
+  private int NUM_NODES = 15;
+
+  public FrequencyDaoImpl(ArrayList<PCAPdata> packets) {
+    allPCAP = packets; 
+  }
+
+  /* Retrieve all necessary entities, processes, and puts into final arraylist */
+  public void loadFrequency() {
+    findMyIP();
+    processData();
+    putFinalFreq();
+  }
+
+  public ArrayList<PCAPdata> getAllPCAP() {
+    return allPCAP;
+  }
+
+  /* Find local ip address based on highest recurring IP address */
+  private void findMyIP() {
+    HashMap<String, Integer> hm = new HashMap<String, Integer>();
+    for (PCAPdata packet : allPCAP) {
+      // source
+      if (hm.containsKey(packet.source)) { 
+        // if IP already exists, increment
+        hm.merge(packet.source, 1, Integer::sum);
+      }
+      else {
+        hm.put(packet.source, 1);
+      }
+      // destination
+      if (hm.containsKey(packet.destination)) { 
+        // if IP already exists, increment
+        hm.merge(packet.destination, 1, Integer::sum);
+      }
+      else {
+        hm.put(packet.destination, 1);
+      }
+    }
+    // find largest recurrence
+    myip = Collections.max(hm.entrySet(), Map.Entry.comparingByValue()).getKey();
+  }
+
+  public String getMyIP() {
+    return myip;
+  }
+
+  /* Fills out finalMap with frequencies based on IP address only (not protocol). */
+  private void processData(){
+    finalMap = new HashMap<String, PCAPdata>();
+    for (PCAPdata packet : allPCAP) {
+      String srcip = packet.source;
+      String dstip = packet.destination;
+      String protocol = "";
+      String outip = "";
+
+      if (srcip.equals(myip)) {
+        outip = dstip;
+      }
+      else {
+        outip = srcip;
+      }
+      
+      // PCAPdata takes in (source, destination, domain, location, protocol, size, flagged, frequency) 
+      if (finalMap.containsKey(outip)){
+        // retrieve current value with outip and increments frequency
+        PCAPdata currPCAP = finalMap.get(outip);
+        currPCAP.incrementFrequency();
```

Comment:
Since you only need frequencies you can just store the count directly instead of storing the PCAP object
```
finalMap = new HashMap<String, Integer>();
finalMap.merge(outip, 1, Integer::sum);
```